### PR TITLE
Task for add search fields

### PR DESCRIPTION
**Pull request by YaroslavSemeniuk**
created a task to add search fields that will be used by the text index

### CODE REVIEW

File: parsers/appendObjectParser.js [hunk 1]
```diff
@@ -35,16 +57,19 @@ const appendObject = async (data, operation) => {
...
+    const { result: isAddedField, error } = await Wobj.addField(data);
+    if (error) throw error;
 
-    if (error) {
-      throw error;
+    let isAddedSearchField = true;
+    if (!_.isEmpty(data.search)) {
+      const { result, error: err } = await Wobj.addSearchField(data);
+      if (error) throw err;
+      isAddedSearchField = result;
```
Comment (Waiviogit) [inline parsers/appendObjectParser.js:67]: why update here? updateSpecificFieldsHelper.update

File: utilities/tasks/addSearchesFieldsToWobjects/addSearchesFieldsToWobjects.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    if (err) {
+      console.error(`Failed to parse the address for the object: ${wobj.author_permlink} with error:`, err);
+      continue;
+    }
+    await WObject.updateOne({ _id: wobj._id }, { $set: { [`search.${fieldName}`]: newFields } });
+  }
+};
+
+const getFieldsData = async (fields, fieldNameForParse) => {
+  const newFields = [];
```
Comment (Waiviogit) [inline utilities/tasks/addSearchesFieldsToWobjects/addSearchesFieldsToWobjects.js:20]: .chain().filter().map()
create object to store callbacks for map instead switch case

File: utilities/tasks/addSearchesFieldsToWobjects/addSearchesFieldsToWobjects.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      case FIELDS_NAMES.NAME:
+      case FIELDS_NAMES.EMAIL:
+        newFields.push(_.get(field, 'body'));
+        break;
+      case FIELDS_NAMES.PHONE:
+        newFields.push(_.get(field, 'number') || _.get(field, 'body'));
+        break;
+      case FIELDS_NAMES.ADDRESS:
+        const { address, err } = parseAddress(_.get(field, 'body'));
+        if (err) return { err };
```
Comment (Waiviogit) [inline utilities/tasks/addSearchesFieldsToWobjects/addSearchesFieldsToWobjects.js:34]: what will be here???

File: validator/appendObjectValidator.js [hunk 4]
```diff
@@ -44,6 +45,15 @@ const validateSameFields = async (data) => {
...
 
+const validateSearchFields = async (data) => {
+  const { wobject } = await Wobj.getOne({ author_permlink: data.author_permlink });
+  if (!_.isEmpty(data.search)) {
+    const searchWords = [].concat(...Object.values(wobject.search));
+    const existWords = _.filter(data.search, (field) => _.includes(searchWords, field));
+    if (!_.isEmpty(existWords)) throw new Error("Can't append word, append is now exist!");
+  }
+};
+
```
Comment (Waiviogit) [inline validator/appendObjectValidator.js:56]: you don't need this if you use $addToSet

**Review by Waiviogit — COMMENTED** [review]
add author_permlink search when creating, add tests

File: parsers/appendObjectParser.js [hunk 5]
```diff
@@ -35,16 +36,18 @@ const appendObject = async (data, operation) => {
       `${data.field.author}_${data.field.permlink}`,
       data.author_permlink,
     );
-    const { result, error } = await Wobj.addField(data);
+    const { result: isAddedField, error } = await Wobj.addField(data);
+    if (error) throw error;
+
+    const {
+      result: isAddedSearchField, error: err,
+    } = await updateSpecificFieldsHelper.addSearchField(data);
```
Comment (Fesmofet) [inline parsers/appendObjectParser.js:44]: await updateSpecificFieldsHelper.addSearchField({...data, searchField: parseSearchData(metadata) });

File: utilities/helpers/updateSpecificFieldsHelper.js [hunk 6]
```diff
@@ -242,4 +243,54 @@ const setMapToChildren = async (authorPermlink, map) => {
...
+        const { address, err } = parseAddress(_.get(metadata, 'wobj.field.body', ''));
+        if (err) return { err };
+        searchField = address;
+        break;
+    }
+    return searchField;
+  }
+};
+
+const addSearchField = async (data) => {
```
Comment (Fesmofet) [inline utilities/helpers/updateSpecificFieldsHelper.js:268]: { author_permlink, field,  searchField}

**Review by Fesmofet — APPROVED** [review]

**Pull request merged**